### PR TITLE
XL8-1459 ts/js based e2e web client

### PR DESCRIPTION
**Pull request by 0hoo**
run yarn -> yarn build -> open index.html. 

We still have a CORS issue, for now pls try it on no CORS mode chrome (open -n -a /Applications/Google\ Chrome.app/Contents/MacOS/Google\ Chrome --args --user-data-dir="/tmp/chrome_dev_test" --disable-web-security)

### CODE REVIEW

**Review by dasony — APPROVED** [review]
LGTM. 

BTW I added a CORS middleware to the gateway server, and it seems to be working at least from my local machine. 

Also Amazon STT seems very slow here, with latency like 30 seconds or more, and it seems to stop working during a session. I will investigate more what's causing it. It could be because it is sending the audio frames too slowly, or sometimes it send them out of order. We might need to improve that, or maybe consider using websocket instead. I will let you know when I find out more.

**Review by jinhyung — APPROVED** [review]
LGTM

File: web/api.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  data: {
+    text: string;
+    original: string;
+    time_start_msec: string;
+    time_end_msec: string;
+    is_partial: boolean;
+  };
+}
+
+const API_ENDPOINT = "http://3.91.88.129/v1/e2e/";
```
Comment (tim-xl8) [inline web/api.ts:18]: Shall we take the endpoint hostname as a parameter or configuration, instead of hard-coding here?

File: web/app.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
+import RecordRTC, { StereoAudioRecorder } from "recordrtc";
+import { initE2E, transE2E } from "./api";
+import { findLastMessageElement, appendMessage, scrollMessages } from "./html";
+import { blobToBase64 } from "./utils";
+
+type onMicrophone = (mic: MediaStream) => void;
+
+const CHUNK = 4096;
+const PERSON_IMG = "https://image.flaticon.com/icons/svg/145/145867.svg";
```
Comment (tim-xl8) [inline web/app.ts:9]: Please feel free to make the Look&Feel of the UI as necessary (maybe in the future). I literally copy&pasted the UI from a chat demo.

**Review by tim-xl8 — COMMENTED** [review]
LGTM!

**Review by tim-xl8 — APPROVED** [review]

**Pull request merged**